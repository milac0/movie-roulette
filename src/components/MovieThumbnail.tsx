import React from "react";
import { css } from "emotion";
import { colors } from "../theme/theme";
import { Movie } from "../interface";
import Hover from "./Hover";
import { getYear } from "../helpers";

const movie = css`
  background: ${colors.backgroundGrey};
  height: 330px;
  position: relative;
  margin-top: -4px;
  position: relative;
  width: 220px;

  .rating {
    background: ${colors.secondary};
    border-radius: 50%;
    color: #000;
    height: 40px;
    position: absolute;
    right: 0.5em;
    text-align: center;
    top: 0.5em;
    width: 40px;

    h1 {
      font-size: 1rem;
      font-weight: 700;
    }
  }

  img {
    height: auto;
    width: 100%;
  }
`;

const hoverDiv = css`
  color: ${colors.primary};
  font-size: 1.5rem;
  font-weight: 700;
  position: relative;
  height: 330px;
  text-align: center;
  width: 220px;

  .title {
    margin: 0;
    padding: 3em 0.5em 0 0.5em;
  }

  .year {
    margin-top: 0;
  }

  .lang {
    bottom: 1em;
    left: 50%;
    margin: 0;
    position: absolute;
    transform: translate(-50%, -50%);
  }
`;

interface Props {
  movie: Movie;
}

const MovieThumbnail: React.FC<Props> = ({
  movie: { poster_path, vote_average, title, original_language, release_date }
}) => {
  const hover = (
    <div className={hoverDiv}>
      <p className="title">{title.toUpperCase() || null}</p>
      <p className="year">{getYear(release_date) || null}</p>
      <p className="lang">{original_language.toUpperCase() || null}</p>
    </div>
  );
  return (
    <Hover onHover={hover}>
      <div className={movie}>
        <div className="rating">
          <h1>{vote_average}</h1>
        </div>
        <img src={`https://image.tmdb.org/t/p/w500${poster_path}`} />
      </div>
    </Hover>
  );
};

export default MovieThumbnail;
