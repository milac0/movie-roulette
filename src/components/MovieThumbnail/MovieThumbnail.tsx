import React from "react";
import Movie from "../../interface/Movie";
import Hover from "../Hover/Hover";
import { getYear } from "../../helpers";
import questionImg from "../../assets/images/question.png";
import styles from "./movieThumbnail.scss";

interface Props {
  movie: Movie;
}

const MovieThumbnail: React.FC<Props> = ({
  movie: { poster_path, vote_average, title, original_language, release_date }
}) => {
  const hover = (
    <div className={styles.hoverDiv}>
      <h1>{title.toUpperCase() || null}</h1>
      <h2>{getYear(release_date) || null}</h2>
      <h3>{original_language.toUpperCase() || null}</h3>
    </div>
  );
  return (
    <Hover onHover={hover}>
      <div className={styles.movie}>
        <div className={styles.rating}>
          <h1>{vote_average}</h1>
        </div>
        {poster_path ? (
          <img src={`https://image.tmdb.org/t/p/w500${poster_path}`} />
        ) : (
            <div className={styles.altContainer}>
              <img src={questionImg} />
            </div>
          )}
      </div>
    </Hover>
  );
};

export default MovieThumbnail;
