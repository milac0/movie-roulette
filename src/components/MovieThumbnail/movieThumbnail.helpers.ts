import Genre from "../../interface/Genre";
import Company from "../../interface/Company";

export const getGenres = (arr: Array<Genre>) => {
    const genres: Array<string> = [];
    arr.slice(0, 3).forEach((x: Genre) => {
        genres.push(x.name);
    });
    return genres.join(", ");
};

export const getRuntime = (duration: number): string =>
    `${Math.floor(duration / 60)}h ${duration % 60}m`;

export const getYear = (date: string) => new Date(date).getFullYear();

export const getProductionCompanies = (companies: Array<Company>) => {
    const list: Array<string> = [];
    companies.forEach(company => list.push(company.name));
    return list.join(", ");
};