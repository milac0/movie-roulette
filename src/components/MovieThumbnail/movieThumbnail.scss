@import "../../theme/variables";

.hoverDiv {
  color: $primary;
  font-size: 1.5rem;
  font-weight: 700;
  position: relative;
  height: 330px;
  text-align: center;
  width: 220px;

  h1 {
    font-size: inherit;
    margin: 0;
    padding: 3em 0.5em 0 0.5em;
  }

  h2 {
    font-size: inherit;
    margin-top: 0;
  }

  h3 {
    bottom: 1em;
    font-size: inherit;
    left: 50%;
    margin: 0;
    position: absolute;
    transform: translate(-50%, -50%);
  }
}

.movie {
  background: $backgroundGrey;
  height: 330px;
  position: relative;
  margin-top: 0;
  position: relative;
  width: 220px;

  .rating {
    background: $secondary;
    border-radius: 50%;
    color: #000;
    height: 40px;
    position: absolute;
    right: 0.5em;
    text-align: center;
    top: 0.5em;
    width: 40px;

    h1 {
      font-size: 1rem;
      font-weight: 700;
    }
  }

  .altContainer {
    align-items: center;
    display: flex;
    height: 330px;
  }

  img {
    height: auto;
    width: 100%;
  }
}
