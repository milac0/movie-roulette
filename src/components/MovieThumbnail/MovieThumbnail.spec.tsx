import React from 'react';
import renderer from 'react-test-renderer';

import MovieThumbnail from './MovieThumbnail';

describe('MovieThumbnail renders correctly', () => {
    test('snapshot renders', () => {
        const movie = {
            popularity: 559.414,
            id: 419704,
            genres: [{
                id: 878,
                name: "Science Fiction"
            }],
            production_companies: [{
                id: 490,
                name: "New Regency Productions",
                origin_country: ""
            }],
            video: false,
            vote_count: 1843,
            vote_average: 6,
            title: "Ad Astra",
            release_date: "2019-09-17",
            original_language: "en",
            original_title: "Ad Astra",
            runtime: 123,
            genre_ids: [878, 18, 53, 12, 9648],
            backdrop_path: "/5BwqwxMEjeFtdknRV792Svo0K1v.jpg",
            adult: false,
            overview: "The near future, a time when both hope and hardships drive humanity to look to the stars and beyond.",
            poster_path: "/xBHvZcjRiWyobQ9kxBhO6B2dtRI.jpg"
        }

        const tree = renderer.create(<MovieThumbnail movie={movie} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});

