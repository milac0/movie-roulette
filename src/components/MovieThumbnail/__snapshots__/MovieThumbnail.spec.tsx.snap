// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieThumbnail renders correctly snapshot renders 1`] = `
<div
  className="hoverCard"
>
  <div
    className="noHover"
  >
    <div
      className="movie"
    >
      <div
        className="rating"
      >
        <h1>
          6
        </h1>
      </div>
      <img
        src="https://image.tmdb.org/t/p/w500/xBHvZcjRiWyobQ9kxBhO6B2dtRI.jpg"
      />
    </div>
  </div>
  <div
    className="hover"
  >
    <div
      className="hoverDiv"
    >
      <h1>
        AD ASTRA
      </h1>
      <h2>
        2019
      </h2>
      <h3>
        EN
      </h3>
    </div>
  </div>
</div>
`;
